### PR TITLE
Log exceptions in the netty channel pipeline

### PR DESCRIPTION
**Pull request by bernd**
This will help debugging channel handler problems in the future.

### CODE REVIEW

File: graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/inputs/transports/NettyTransport.java [hunk 1]
```diff
@@ -192,6 +193,20 @@ public void stop() {
...
         LinkedHashMap<String, Callable<? extends ChannelHandler>> handlerList = Maps.newLinkedHashMap();
 
+        handlerList.put("exception-logger", new Callable<ChannelHandler>() {
+            @Override
+            public ChannelHandler call() throws Exception {
+                return new SimpleChannelUpstreamHandler() {
+                    @Override
+                    public void exceptionCaught(ChannelHandlerContext ctx, ExceptionEvent e) throws Exception {
+                        log.error("Error on Input [" + input.getName() + "/" + input.getId() + "] (channel "
+                                + e.getChannel().toString() + ")", e.getCause());
```
Comment (joschi) [inline graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/inputs/transports/NettyTransport.java]: Would it make sense to log the inner most exception (with [`Throwables#getRootCause(java.lang.Throwable)`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Throwables.html#getRootCause(java.lang.Throwable)))?

Comment (bernd) [inline graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/inputs/transports/NettyTransport.java]: In #1105 the error message looked like this:

```
2015-08-04 17:30:37,729 ERROR: org.graylog2.plugin.inputs.transports.NettyTransport - Error on Input [Syslog TCP/55c0b1cf0c744fd8e91c8958] (channel [id: 0xa6d26be2, /10.0.0.1:34500 :> /10.0.0.2:1514])
java.lang.IndexOutOfBoundsException: Too many bytes to read - Needs 16450, maximum is 16446 or 171
	at org.jboss.netty.buffer.CompositeChannelBuffer.getBytes(CompositeChannelBuffer.java:232)
	at org.jboss.netty.buffer.HeapChannelBuffer.setBytes(HeapChannelBuffer.java:127)
	at org.jboss.netty.buffer.AbstractChannelBuffer.writeBytes(AbstractChannelBuffer.java:467)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.extractFrame(FrameDecoder.java:567)
	at org.graylog2.inputs.syslog.tcp.SyslogOctetCountFrameDecoder.decode(SyslogOctetCountFrameDecoder.java:55)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.callDecode(FrameDecoder.java:425)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.messageReceived(FrameDecoder.java:310)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564)
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791)
	at org.graylog2.inputs.syslog.tcp.SyslogTCPFramingRouterHandler.messageReceived(SyslogTCPFramingRouterHandler.java:53)
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:70)
[...]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
2015-08-04 17:30:37,729 ERROR: org.graylog2.plugin.inputs.transports.NettyTransport - Error in channel [id: 0xa6d26be2, /10.0.0.1:34500 :> /10.0.0.2:1514]
java.lang.NumberFormatException: For input string: "<45>1"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Integer.parseInt(Integer.java:492)
	at java.lang.Integer.parseInt(Integer.java:527)
	at org.graylog2.inputs.syslog.tcp.SyslogOctetCountFrameDecoder.decode(SyslogOctetCountFrameDecoder.java:44)
```

The first exception actually helped to debug the problem so I guess it's fine as it is.

Comment (joschi) [inline graylog2-plugin-interfaces/src/main/java/org/graylog2/plugin/inputs/transports/NettyTransport.java]: :white_check_mark:

**Comment by joschi** [comment]
LGTM. :shipit:

**Pull request merged**